### DISCUSSION TITLE
近期两天对mtkwifi的更新貌似导致了ssrplus对无线的代理超时

### DISCUSSION THREAD

**Discussion by w311ang** [coolsnowwolf/lede#10026] [Bugs / 问题反馈]
我连K2P无线用`curl 1.1.1.1`，测了一下是超时了，如果是直连应该是返回
```html
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>cloudflare</center>
</body>
</html>
```
所以应该是被redirect到端口1234了，不然应该会直连，但是用有线连又是可以正常代理打开google的
https://github.com/fw876/helloworld/issues/975

**Comment by w311ang** [comment]
再补充一点，连无线是可以访问局域网电脑的，看起来又不像是被vlan隔离了

**Comment by w311ang** [comment]
看了下commits，怀疑一下https://github.com/coolsnowwolf/lede/commit/a28334d8b8dfa58dccd6fce369f5ee87e5b5a703

**Reply by w311ang** [reply]
一键回滚
```bash
changed_files=$(git diff-tree --no-commit-id --name-only -r a28334d)

while IFS= read -r line; do
    git checkout a28334d~1 -- $line
done <<< "$changed_files"
```

**Comment by w311ang** [comment]
拿之前打patch编译正常的版本再次编译果然出了问题，应该就是最近更新搞的了

**Comment by w311ang** [comment]
这是[编辑战](https://zh.m.wikipedia.org/zh-cn/Wikipedia:%E7%B7%A8%E8%BC%AF%E6%88%B0)了属于是，前面的我应该猜错了，这次应该准了
https://github.com/coolsnowwolf/lede/blame/1ccc3bfd54f7d9a2a2a02f2cb2b062452ea8d830/target/linux/ramips/dts/mt7621.dtsi#L560-L561
https://github.com/coolsnowwolf/lede/blame/a28334d8b8dfa58dccd6fce369f5ee87e5b5a703/target/linux/ramips/dts/mt7621.dtsi#L566-L567
https://github.com/coolsnowwolf/lede/blame/14e109d0b91f61f910a5a919a0440468b3e15161/target/linux/ramips/dts/mt7621.dtsi#L566-L567

**Comment by w311ang** [comment]
#10030 fixed

**Comment by sdf8057** [comment]
👍🏻

**Comment by w311ang** [comment]
好像commit被撤回了，想修复的自己应用patch吧
```diff
--- target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:49:48.820003170 +0800
+++ target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:53:12.850003092 +0800
@@ -557,8 +557,8 @@
 
 	hnat: hnat@1e100000 {
 		compatible = "mediatek,mtk-hnat_v1";
-		ext-devices = "apcli0", "apclii0","apclix0",
-			"ra0", "rai0", "rax0", "wlan0", "wlan1";
+		/* ext-devices = "apcli0", "apclii0","apclix0",
+			"ra0", "rai0", "rax0", "wlan0", "wlan1"; */
 		reg = <0x1e100000 0x3000>;
 
 		resets = <&ethsys 0>;
```

**Reply by MeIsReallyBa** [reply]
> 好像commit被撤回了，想修复的自己应用patch吧
> 
> ```diff
> --- target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:49:48.820003170 +0800
> +++ target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:53:12.850003092 +0800
> @@ -557,8 +557,8 @@
>  
>  	hnat: hnat@1e100000 {
>  		compatible = "mediatek,mtk-hnat_v1";
> -		ext-devices = "apcli0", "apclii0","apclix0",
> -			"ra0", "rai0", "rax0", "wlan0", "wlan1";
> +		/* ext-devices = "apcli0", "apclii0","apclix0",
> +			"ra0", "rai0", "rax0", "wlan0", "wlan1"; */
>  		reg = <0x1e100000 0x3000>;
>  
>  		resets = <&ethsys 0>;
> ```

启用whnat后，就不需要注释这个了。这个注释后，wifi就是纯cpu软转发。

**Reply by w311ang** [reply]
> > 好像commit被撤回了，想修复的自己应用patch吧
> > ```diff
> > --- target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:49:48.820003170 +0800
> > +++ target/linux/ramips/dts/mt7621.dtsi	2022-08-31 23:53:12.850003092 +0800
> > @@ -557,8 +557,8 @@
> >  
> >  	hnat: hnat@1e100000 {
> >  		compatible = "mediatek,mtk-hnat_v1";
> > -		ext-devices = "apcli0", "apclii0","apclix0",
> > -			"ra0", "rai0", "rax0", "wlan0", "wlan1";
> > +		/* ext-devices = "apcli0", "apclii0","apclix0",
> > +			"ra0", "rai0", "rax0", "wlan0", "wlan1"; */
> >  		reg = <0x1e100000 0x3000>;
> >  
> >  		resets = <&ethsys 0>;
> > ```
> 
> 启用whnat后，就不需要注释这个了。这个注释后，wifi就是纯cpu软转发。

非常感谢，我一直以为是代理跟hnat冲突，那我肯定选代理，没想到开启whnat就都可以兼得了

**Comment by MeIsReallyBa** [comment]
需要打开whnat，wifi走hnat才能正常redirect。
https://github.com/coolsnowwolf/lede/blob/655566bebf608d1ca9d6b447f2a1edf64ad402e3/target/linux/ramips/files/drivers/net/ethernet/mtk/mtk_hnat/hnat.c#L858

**Reply by w311ang** [reply]
```diff
--- target/linux/ramips/files/drivers/net/ethernet/mtk/mtk_hnat/hnat.c	2022-09-01 18:09:41.958660156 +0800
+++ target/linux/ramips/files/drivers/net/ethernet/mtk/mtk_hnat/hnat.c	2022-09-01 18:10:19.308660141 +0800
@@ -855,7 +855,7 @@ static int hnat_remove(struct platform_d
 
 static const struct mtk_hnat_data hnat_data_v1 = {
 	.num_of_sch = 2,
-	.whnat = false,
+	.whnat = true,
 	.per_flow_accounting = false,
 	.mcast = false,
 	.version = MTK_HNAT_V1,
```

**Reply by WYC-2020** [reply]
那有没办法走不走这个都可以，能优化的话尽可能不受这个影响

**Reply by MeIsReallyBa** [reply]
> 那有没办法走不走这个都可以，能优化的话尽可能不受这个影响

只有打开whnat，无线offload才不会和代理冲突。暂时没找到其他解决办法。
